src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -118,9 +124,53 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         this.exercise.participationStatus = participationStatus(this.exercise);
         this.isAfterAssessmentDueDate = !this.exercise.assessmentDueDate || moment().isAfter(this.exercise.assessmentDueDate);
         this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.exercise);
+        if (this.exercise.type.toString() === 'programming') {
+            this.getProgrammingExercise();
+        }
         this.subscribeForNewResults();
     }
 
+    /**
+     * The button should be only visible in the dev environment
+     */
+    isInProduction(): boolean {
+        return this.profileService.isInProduction();
+    }
+
+    /**
+     * checks if exercise has no local setup
+     */
+    hasNoLocalSetup(): boolean {
+        return this.programmingExerciseSimulationUtils.hasNoLocalSetup(this.programmingExercise.testRepositoryUrl);
+    }
+
+    /**
+     * asks the server for the programming exercise with the provided exercise ID
+     */
+    getProgrammingExercise() {
+        this.courseExerciseSubmissionResultSimulationService.getProgrammingExercise(this.exerciseId).subscribe((programmingExercise) => {
+            this.programmingExercise = programmingExercise;
+        });
+    }
+
+    /**
+     * triggers the simulation of a participation and submission for the currently logged in user
+     */
+    simulateSubmission() {
+        this.courseExerciseSubmissionResultSimulationService.simulateSubmission(this.exerciseId).subscribe(() => {
+            this.jhiAlertService.success('artemisApp.exercise.submissionSuccessful');
```

Comment:
The alert service needs to be injected into the component, otherwise it will be undefined as can be seen in this console error when clicking on "Submission" or "Result":

![OnPaste 20200413-001436](https://user-images.githubusercontent.com/7109225/79081487-316ded00-7d1e-11ea-8ee6-6c3bda2c4e89.png)

Reply:
i added the injection